tests/integration/targets/copr/tasks/main.yml

```diff
@@ -0,0 +1,64 @@
+---
+- block:
+  - name: enable copr project
+    copr:
+      host: copr.fedorainfracloud.org
+      state: enabled
+      name: '@copr/integration_tests'
+      chroot: centos-stream-x86_64
+    register: result
+
+  - name: assert that the copr project was enabled
+    assert:
+      that:
+        - 'result is changed'
+        - result.msg == 'enabled'
+        - result.info == 'Please note that this repository is not part of the main distribution'
+
+  - name: enable copr project
+    check_mode: yes
+    copr:
+      state: enabled
+      name: '@copr/integration_tests'
+      chroot: centos-stream-x86_64
+    register: result
+
+  - name: assert that the copr project was enabled
+    assert:
+      that:
+        - result.state == 'enabled'
+        - result.chroot == 'centos-stream-x86_64'
+        - result.host == 'copr.fedorainfracloud.org'
+
+  - name: remove copr project
+    copr:
+      state: absent
+      name: '@copr/integration_tests'
+    register: result
+
+  - name: assert that the copr project was enabled
+    assert:
+      that:
+        - 'result is changed'
+        - result.msg == 'absent'
+
+  - name: disable copr project
+    copr:
+      state: disabled
+      name: '@copr/integration_tests'
+      chroot: centos-stream-x86_64
+    register: result
+
+  - name: assert that the copr project was disabled
+    assert:
+      that:
+        - 'result is changed'
```

Comment:
Shouldn't this be `result is not changed`? Is it because of the value for `chroot`? Or is the module not idempotent?

Reply:
It works so that the user does not have a repository on the host, then runs the task with state disabled and the repository appears on the host but is set to disable, so a change will occur. The module is idempotent.